server/src/models/swing.rs

```diff
@@ -110,3 +278,332 @@ fn decode(model: &[u8]) -> (f64, f64) {
         )
     }
 }
+
+/// Calculate `a` and `b` of a linear function that intersects with the data
+/// points (first_timestamp, first_value) and (final_timestamp, final_value).
+fn calculate_a_and_b(
+    first_timestamp: Timestamp,
+    first_value: f64,
+    final_timestamp: Timestamp,
+    final_value: f64,
+) -> (f64, f64) {
+    // It seems that no values can be assigned to first_value and final_value
+    // so a * timestamp + b = INFINITY and a * timestamp + b = NEG_INFINITY.
```

Comment:
I do not quite understand this comment, what is its purpose? Should it be moved down to the else clause?